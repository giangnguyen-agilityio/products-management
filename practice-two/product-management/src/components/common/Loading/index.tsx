// Libraries
import { memo } from 'react'
import { Container, Flex, Spinner, Text } from '@chakra-ui/react'

const Loading = () => {
  return (
    <Container>
      <Flex
        justifyContent="center"
        flexDirection="column"
        margin="0 auto"
        height="100vh"
        borderRadius={10}
        fontFamily="Oswald-Regular"
        alignItems="center"
      >
        {/* Display a spinning loading indicator */}
        <Spinner
          thickness="4px"
          speed="0.65s"
          emptyColor="gray.200"
          size="xl"
        />
        {/* Display a text indicating the loading process */}
        <Text
          variant="fontSecondary"
          fontSize="lg"
          color="secondary"
          marginTop={2}
        >
          Loading....
        </Text>
      </Flex>
    </Container>
  )
}

export default memo(Loading)
