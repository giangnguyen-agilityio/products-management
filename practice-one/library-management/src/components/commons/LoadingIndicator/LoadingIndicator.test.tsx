import React from 'react'
import {render} from '@testing-library/react'
import Loading from '.'

describe('Loading component', () => {
  it('should render the loading indicator when isLoading is true', () => {
    const {container} = render(<Loading isLoading={true} />)
    expect(container).toMatchSnapshot()
  })

  it('should not render the loading indicator when isLoading is false', () => {
    const {container} = render(<Loading isLoading={false} />)
    expect(container).toMatchSnapshot()
  })

  it('should have the correct CSS class when isLoading is true', () => {
    const {container} = render(<Loading isLoading={true} />)
    const loadingIndicator = container.querySelector('.loading-indicator')

    expect(loadingIndicator).toHaveClass('loading-indicator')
  })

  it('should render the correct number of wave elements when isLoading is true', () => {
    const {container} = render(<Loading isLoading={true} />)
    const waveElements = container.querySelectorAll('.wave')

    expect(waveElements.length).toBe(6)
  })

  it('should not render any wave elements when isLoading is false', () => {
    const {container} = render(<Loading isLoading={false} />)
    const waveElements = container.querySelectorAll('.wave')

    expect(waveElements.length).toBe(0)
  })
})
