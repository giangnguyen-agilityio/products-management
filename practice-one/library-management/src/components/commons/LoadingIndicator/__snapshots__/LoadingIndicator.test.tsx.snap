// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should not render the loading indicator when isLoading is false 1`] = `<div />`;

exports[`Loading component should render the loading indicator when isLoading is true 1`] = `
<div>
  <div
    class="loading-indicator"
  >
    <ul
      class="loading-icon"
    >
      <li
        class="wave"
      />
      <li
        class="wave"
      />
      <li
        class="wave"
      />
      <li
        class="wave"
      />
      <li
        class="wave"
      />
      <li
        class="wave"
      />
    </ul>
  </div>
</div>
`;
